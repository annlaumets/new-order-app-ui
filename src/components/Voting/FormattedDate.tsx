import { Flex, Text } from '@chakra-ui/react'
import dayjs from 'dayjs'

interface FormattedDateProps {
  date: number
}
const FormattedDate: React.FC<FormattedDateProps> = ({ date }) => {
  return (
    <Flex alignItems="center" gap="2">
      <Text fontSize="x-small" color="white">
        {dayjs.unix(date).format('MM.DD.YYYY')}
      </Text>
      <Text fontSize="x-small" color="gray.50">
        {dayjs.unix(date).format('HH.mm')}
      </Text>
    </Flex>
  )
}

export default FormattedDate
