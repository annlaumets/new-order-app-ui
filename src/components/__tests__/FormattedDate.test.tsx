import { render } from '@testing-library/react'
import dayjs from 'dayjs'

import FormattedDate from 'components/Voting/FormattedDate'

describe('FormattedDate.tsx component', () => {
  const { container } = render(
    <FormattedDate date={dayjs(new Date('2022-01-01')).startOf('day').unix()} />
  )

  test('match to snapshot', () => {
    expect(container).toMatchSnapshot()
  })
})
