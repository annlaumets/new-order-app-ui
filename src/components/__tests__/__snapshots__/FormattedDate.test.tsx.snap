// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedDate.tsx component match to snapshot 1`] = `
<div>
  <div
    class="css-j2at52"
  >
    <p
      class="chakra-text css-19j6lia"
    >
      01.01.2022
    </p>
    <p
      class="chakra-text css-3yq4qz"
    >
      00.00
    </p>
  </div>
</div>
`;
